In [1]:
import joblib
from DbConn import *
import pandas as pd
import numpy as np

db = DbConn()

# 총 관객수 예측 모델
model = joblib.load('C:/Users/user/SweetBox/test/max_people.pkl')

In [11]:
# 예측할 영화 정보 초기 값
MOVIE_NAME = "극한직업"
PRODUCER = "(주)어바웃필름"
DIRECTOR = "봉준호"
ACTOR1 = "송강호"
ACTOR2 = "이선균"
ACTOR3 = "조여정"
ACTOR4 = "최우식"
SCREEN = "1553"

MOVIE_NAME = "기생충"
PRODUCER = "(주)바른손씨앤에이"
DIRECTOR = "이병헌"
ACTOR1 = "류승룡"
ACTOR2 = "이하늬"
ACTOR3 = "진선규"
ACTOR4 = "이동휘"
SCREEN = "1783"

MOVIE_NAME = "테넷"
PRODUCER = "워너 브러더스 픽쳐스"
DIRECTOR = "크리스토퍼 놀란"
ACTOR1 = "존 데이비드 워싱턴"
ACTOR2 = "로버트 패틴슨"
ACTOR3 = "엘리자베스 데비카"
ACTOR4 = "마이클 케인"
SCREEN = "2228"

MOVIE_NAME = "도굴"
PRODUCER = "(주)싸이런픽쳐스"
DIRECTOR = "박정배"
ACTOR1 = "이제훈"
ACTOR2 = "조우진"
ACTOR3 = "신혜선"
ACTOR4 = "임원희"
SCREEN = "1000"

MOVIE_NAME = "반도"
PRODUCER = "(주)영화사레드피터"
DIRECTOR = "연상호"
ACTOR1 = "강동원"
ACTOR2 = "이정현"
ACTOR3 = "이레"
ACTOR4 = "권해효"
SCREEN = "2490"

MOVIE_NAME = "다만 악에서 구하소서"
PRODUCER = "(주)하이브미디어코프"
DIRECTOR = "홍원찬"
ACTOR1 = "황정민"
ACTOR2 = "이정재"
ACTOR3 = "박정민"
ACTOR4 = "박소이"
SCREEN = "1837"

MOVIE_NAME = "스파이더맨 파프롬홈"
PRODUCER = "소니 픽쳐스 애니메이션"
DIRECTOR = "존 왓츠"
ACTOR1 = "톰 홀랜드"
ACTOR2 = "사무엘 L. 잭슨"
ACTOR3 = "젠다야 콜맨"
ACTOR4 = "코비 스멀더스"
SCREEN = "1944"

# parameters ----------------

# producer
PROD_MAX = 0
PROD_AVG = 0
prod_sql = """
SELECT max(MAX_AUDIACC) AS PROD_MAX, avg(MAX_AUDIACC) AS PROD_AVG 
FROM MOVIE2 
WHERE PRODNM = :PRODUCER
"""
prod_res = db.execute(prod_sql, {"PRODUCER":PRODUCER})
if prod_res[0][0] != None :
    PROD_MAX = prod_res[0][0]
if prod_res[0][1] != None :
    PROD_AVG = prod_res[0][1]
    
# director
DIRECTOR_AVG = 0
DIRECTOR_MAX = 0
DIRECTOR_100CNT = 0
director_sql = """
SELECT ROUND(max(MAX_AUDIACC)) AS DIRECTOR_MAX, ROUND(avg(MAX_AUDIACC)) AS DIRECTOR_AVG
FROM MOVIE2 m WHERE DIRECTOR = :DIRECTOR
"""
director_res = db.execute(director_sql, {"DIRECTOR":DIRECTOR})
if director_res[0][0] != None :
    DIRECTOR_MAX = director_res[0][0]
    DIRECTOR_AVG = director_res[0][1]

director100_sql = """
SELECT COUNT(*) FROM MOVIE2 m WHERE DIRECTOR = :DIRECTOR AND MAX_AUDIACC > 1000000
"""
director100_res = db.execute(director100_sql, {"DIRECTOR":DIRECTOR})
if director100_res[0][0] != None :
    DIRECTOR_100CNT = director100_res[0][0]


# ACTOR
actor_sql = """
SELECT max(MAX_AUDIACC) FROM MOVIE2 m WHERE ACTORS1 = :ACTORNAME OR ACTORS2 = :ACTORNAME OR ACTORS3 = :ACTORNAME OR ACTORS4 = :ACTORNAME
"""
ACTORS1_MAX = 0
actor1_res = db.execute(actor_sql, {"ACTORNAME":ACTOR1})

if actor1_res[0][0] != None :
    ACTORS1_MAX = actor1_res[0][0]

ACTORS2_MAX = 0
actor2_res = db.execute(actor_sql, {"ACTORNAME":ACTOR2})

if actor2_res[0][0] != None :
    ACTORS2_MAX = actor2_res[0][0]

ACTORS3_MAX = 0
actor3_res = db.execute(actor_sql, {"ACTORNAME":ACTOR3})
if actor3_res[0][0] != None :
    ACTORS3_MAX = actor3_res[0][0]
    
ACTORS4_MAX = 0
actor4_res = db.execute(actor_sql, {"ACTORNAME":ACTOR4})
if actor4_res[0][0] != None :
    ACTORS4_MAX = actor4_res[0][0]

SCRNCNT = int(SCREEN)


params = [
    [PROD_MAX, DIRECTOR_AVG, DIRECTOR_MAX, ACTORS1_MAX, ACTORS2_MAX, ACTORS3_MAX, ACTORS4_MAX, DIRECTOR_100CNT, PROD_AVG, SCRNCNT]
]


print("params:", params)
result = model.predict(params)
print(MOVIE_NAME,"의 예측값:", result[0])

params: [[1019857, 7628379, 8006860, 8006860, 8006860, 8006860, 8006860, 2, 559496.6666666666, 1944]]
스파이더맨 파프롬홈 의 예측값: 7249898


params: [[16252575, 6712176, 10912664, 12166659, 10065245, 10065245, 10065245, 4, 8156843, 1553]]
극한직업 의 예측값: 14401147
    
params: [[0, 6828063, 16252575, 17583608, 16252575, 16252575, 16252575, 3, 0, 1783]]
기생충 의 예측값: 11181178
    
params: [[5818753, 3835751, 10283774, 0, 2651188, 0, 10283774, 5, 1347113.4285714286, 2228]]
테넷 의 예측값: 6573512
    
params: [[3838167, 0, 0, 4101671, 4774549, 0, 2961119, 0, 3838167, 1000]]
도굴 의 예측값: 1001207
    
params: [[11551067, 3171305, 11551067, 9686269, 755371, 2695392, 42158, 1, 4273836.333333333, 2490]]
반도 의 예측값: 11551067
    
params: [[5568719, 437884, 437884, 14241429, 12976528, 3388214, 0, 0, 2167070.3333333335, 1837]]
다만 악에서 구하소서 의 예측값: 1001207

params: [[1019857, 7628379, 8006860, 8006860, 8006860, 8006860, 8006860, 2, 559496.6666666666, 1944]]
스파이더맨 파프롬홈 의 예측값: 7249898